iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -520,25 +521,27 @@ public static List<String> getPartitionNames(Table table, Map<String, String> pa
     }
   }
 
-  public static long getPartitionHash(Table icebergTable, String partitionPath) throws IOException {
-    PartitionsTable partitionsTable = (PartitionsTable) MetadataTableUtils
-        .createMetadataTableInstance(icebergTable, MetadataTableType.PARTITIONS);
 
-    try (CloseableIterable<FileScanTask> fileScanTasks = partitionsTable.newScan().planFiles()) {
-      return FluentIterable.from(fileScanTasks)
-          .transformAndConcat(task -> task.asDataTask().rows())
-          .transform(row -> {
-            StructProjection data = row.get(IcebergTableUtil.PART_IDX, StructProjection.class);
-            PartitionSpec spec = icebergTable.specs().get(row.get(IcebergTableUtil.SPEC_IDX, Integer.class));
-            PartitionData partitionData = IcebergTableUtil.toPartitionData(data,
-                Partitioning.partitionType(icebergTable), spec.partitionType());
-            String path = spec.partitionToPath(partitionData);
-            return Maps.immutableEntry(path, data);
-          })
-          .filter(e -> e.getKey().equals(partitionPath))
-          .transform(e -> IcebergAcidUtil.computeHash(e.getValue()))
-          .get(0);
+  public static Integer getPartitionSpecId(Table icebergTable, String partitionPath) {
```

Comment:
`getSpecIdByPartition(Table icebergTable, String partitionName)` ?

Reply:
Changed the method to return `PartitionSpec`, but it cannot replace `Partitioning.partitionType(icebergTable)`. Because the later returns all partition fields, from all partition specs, which is needed for us, but `getPartitionSpec` returns only one spec, it doesn't have all the fields.

Reply:
Why do we need all specs? We find the spec conforming to the partition pattern, don't we?

Reply:
Because `partition` field in `files` metadata table contains partition struct with all partition fields, from all partition specs, so the condition on partition struct should include all fields.

Reply:
`partition` field in `files` metadata table contains partition struct with all partition fields, from all partition specs, so the condition on partition struct should include all fields, for that we need `Partitioning.partitionType(icebergTable)`

fields we can get from our spec, because all fields not in the spec need to have null values - done.

Reply:
so what, don't you construct the predicate to only check the relevant fields from the relevant PartitionSpec?